.github/validate_robopages.yml

```diff
@@ -0,0 +1,115 @@
+name: Validate Contributions
+
+on:
+  pull_request:
+    paths:
+      - '**.yml'
+      - '!.github/**'
+    branches:
+      - main
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: read
+      #issues: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #v4.2.2
+
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@c47758b77c9736f4b2ef4073d4d51994fabfe349 # 3.7.1
+
+      - name: Validate Contribution Files
+        id: robopages-validation
+        continue-on-error: true
+        run: |
+          validate_file() {
+            local file="$1"
+
+            if [[ ! "$file" =~ ^[a-zA-Z0-9_\-./]+\.yml$ ]]; then
+              echo "Invalid file path characters: $file"
+              return 1
+            }
+
+            if [[ "$file" == *"../"* ]]; then
+              echo "Directory traversal attempt detected: $file"
+              return 1
+            }
+
+            docker pull dreadnode/robopages:latest@sha256:2e355b1c46682af2fb586648016c7a28f113a407763b37c74251ae2050b396bb # f65195e7aaf5
+
+            # Run validation
+            docker run --rm \
+              -v $(pwd):/workspace \
+              -w /workspace \
+              dreadnode/robopages:latest validate --path "$(printf '%q' "$file")"
+          }
+
+          # Get changed files, excluding .github directory
+          changed_files=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | \
+            grep '\.yml$' | grep -v '^.github/' || true)
+
+          # Validate each changed file
+          for file in $changed_files; do
+            echo "Validating $file..."
+            validate_file "$file" || exit 1
+          done
+
+      - name: Fallback YAML validation
```

Comment:
what's the value of this vs what the tool already does?

Reply:
this was a fallback proposal mechanism in-case the docker pull and verify were to fail for whatever reason

i'm happy to remove it, but the idea was based on:
- if a contribution were to be submitted, but was not actually loaded within the tool and hence the validator won't be triggered (ie a random yml is submitted as a PR)

Reply:
if a random yml is submitted as PR i'd say that whether or not it was indented right is not that important, no? :P

Reply:
[chore: simon is right rm format checking](https://github.com/dreadnode/robopages/pull/7/commits/c74f7ea10b39c5621aaa8c6efd3854c3f923b38c)
you are correct, my apologies for overcomplicating this